File: python/resistor-color-duo/resistor_color_duo.py
COLORS = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
]


def value(colors):
    result = ''.join([str(COLORS.index(i)) for i in colors])
    return int(result[0:2])

File: typescript/resistor-color-duo/resistor-color-duo.ts
const COLORS: string[] = [
  'black', 
  'brown', 
  'red', 
  'orange', 
  'yellow', 
  'green', 
  'blue', 
  'violet', 
  'grey', 
  'white'
] 

export class ResistorColor {
  private colors: string[];

  constructor(colors: string[]) {
    this.colors = colors;
    if(this.colors.length <= 1){
      throw new Error("At least two colors need to be present")
    }
  }
  
  value = (): number => Number(
    this.colors
      .slice(0, 2)
      .map(i => COLORS.indexOf(i))
      .join('')
    )
}
